src/connector/src/source/filesystem/opendal_source/opendal_enumerator.rs

```diff
@@ -120,3 +131,55 @@ impl<Src: OpendalSource> OpendalEnumerator<Src> {
     }
 }
 pub type ObjectMetadataIter = BoxStream<'static, ConnectorResult<FsPageItem>>;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::source::filesystem::opendal_source::OpendalS3;
+    use crate::source::filesystem::s3::enumerator::get_prefix;
+
+    fn calculate_list_prefix(prefix: &str) -> String {
+        OpendalEnumerator::<OpendalS3>::extract_list_prefix(prefix)
+    }
+
+    #[test]
+    fn test_prefix_logic() {
+        let test_cases = vec![
+            ("a/b/c/hello*/*.json", "a/b/c/hello", "a/b/c/"),
+            ("a/b/c.json", "a/b/c.json", "a/b/"),
+            ("a/b/c/", "a/b/c/", "a/b/c/"),
+            ("a/b/c", "a/b/c", "a/b/"),
+            ("file.json", "file.json", "/"),
+            ("*.json", "", "/"),
+            ("a/b/c/[h]ello*/*.json", "a/b/c/", "a/b/c/"),
```

Comment:
according to the logic `prefix.rfind('/')`, why this case expects `a/b/c/` rather than `a/b/c/[h]ello/*`